### ISSUE TITLE
Allow modifying userpass policies

### ISSUE THREAD

**Issue by mtougeron** [hashicorp/vault#909]
Once you create a new user using the `userpass` auth method the list of policies essentially becomes static. If I want to remove a policy from an account or, as is more common, add a policy to an account I need to recreate the entire user.

It would be extremely helpful for us to be able to manipulate existing userpass policy associations without having to regenerate a password for the user(s).

Here are the steps I used to reproduce using `vault server -dev`

```
$> export VAULT_ADDR='http://127.0.0.1:8200'

$> vault auth ebf8e8ea-dafa-6b42-ac96-ea99dc261d9e
Successfully authenticated!
token: ebf8e8ea-dafa-6b42-ac96-ea99dc261d9e
token_duration: 0
token_policies: [root]

$> vault auth-enable userpass
Successfully enabled 'userpass' at 'userpass'!

$> cat foo.policy.hcl 

path "auth/*" {
  policy = "read"
}

path "cubbyhole/*" {
  policy = "write"
}

$> vault policy-write foo foo.policy.hcl                        
Policy 'foo' written.

$> cat bar.policy.hcl 

path "secret/*" {
  policy = "write"
}

$> vault policy-write bar bar.policy.hcl
Policy 'bar' written.

$> vault write auth/userpass/users/foo password=bar policies=foo
Success! Data written to: auth/userpass/users/foo

$> vault auth -method=userpass username=foo password=bar
Successfully authenticated!
token: 63b99e5d-5f87-f9bd-24d0-3ed3339d3128
token_duration: 2592000
token_policies: [foo]

$> vault auth ebf8e8ea-dafa-6b42-ac96-ea99dc261d9e
Successfully authenticated!
token: ebf8e8ea-dafa-6b42-ac96-ea99dc261d9e
token_duration: 0
token_policies: [root]

$> vault write auth/userpass/users/foo  policies=foo,bar
Success! Data written to: auth/userpass/users/foo

$> vault auth -method=userpass username=foo password=bar
Error making API request.

URL: PUT http://127.0.0.1:8200/v1/auth/userpass/login/foo
Code: 400. Errors:

* unknown username or password
```

Thanks,
Mike

**Comment by cburroughs** [comment]
@pdbogen suggested in IRC that this might be relatively straightforward and "Just need to patch pathUserWrite in builtin/credential/userpass/path_users.go to retrieve and use the old hash if no password was passed as a parameter"

**Comment by jefferai** [comment]
The proper way to do this would be to use the separate `create`/`update` mechanism introduced in 0.5.

**Comment by jefferai** [comment]
This never got closed, but in 0.5.2 there are explicit paths for updating a user's associated policies and password. See https://www.vaultproject.io/docs/auth/userpass.html
